content/en/docs/setup/install/virtual-machine/index.md

```diff
@@ -182,45 +175,122 @@ Run the following commands on the virtual machine you want to add to the Istio m
+
+    {{< text bash >}}
+    $ sudo  mkdir -p /var/run/secrets/tokens
+    $ sudo cp "${HOME}"/istio-token /var/run/secrets/tokens/istio-token
+    {{< /text >}}
+    
 1. Install the `deb` package containing the Istio virtual machine integration runtime.
 
     {{< text bash >}}
     $ curl -LO https://storage.googleapis.com/istio-release/releases/{{< istio_full_version >}}/deb/istio-sidecar.deb
     $ sudo dpkg -i istio-sidecar.deb
     {{< /text >}}
 
-1. Install `root-cert.pem`, `key.pem` and `cert-chain.pem` within the directory `/etc/certs/`.
+1. Install `cluster.env` within the directory `/var/lib/istio/envoy/`.
 
     {{< text bash >}}
-    $ sudo mkdir -p /etc/certs
-    $ sudo cp {root-cert.pem,cert-chain.pem,key.pem} /etc/certs
+    $ sudo cp "${HOME}"/cluster.env /var/lib/istio/envoy/cluster.env
     {{< /text >}}
 
-1. Install `cluster.env` within `/var/lib/istio/envoy/`.
+1. Install `sidecar.env` within the directory`/var/lib/istio/envoy/`.
 
     {{< text bash >}}
-    $ sudo cp cluster.env /var/lib/istio/envoy
+    $ sudo cp "${HOME}"/sidecar.env /var/lib/istio/envoy/sidecar.env
     {{< /text >}}
 
 1. Add the istiod host to `/etc/hosts`.
 
     {{< text bash >}}
-    $ sudo sh -c 'cat hosts-addendum >> /etc/hosts'
+    $ sudo sh -c 'cat $(eval echo ~$SUDO_USER)/hosts-addendum >> /etc/hosts'
+    {{< /text >}}
+
+1. Install the root certificate in the directory  `/etc/certs`
+
+    {{< text bash >}}
+    $ sudo cp "${HOME}"/root-cert.pem /var/run/secrets/istio/root-cert.pem
     {{< /text >}}
 
 1. Transfer ownership of the files in `/etc/certs/` and `/var/lib/istio/envoy/` to the Istio proxy.
 
     {{< text bash >}}
-    $ sudo chown -R istio-proxy /etc/certs /var/lib/istio/envoy
+    $ sudo mkdir -p /etc/istio/proxy
+    $ sudo chown -R istio-proxy /var/lib/istio /etc/certs /etc/istio/proxy  /var/run/secrets
     {{< /text >}}
 
-1. Start Istio within the virtual machine.
+## Setup policies for virtual machine
+1. Get the external ip or internal ip of the virtual machine and store to `"${VM_IP}"`
 
     {{< text bash >}}
-    $ sudo systemctl start istio
+    $ VM_IP="<the external ip or internal ip of the virtual machine>"
     {{< /text >}}
+    
+1. Create a new service
+
+    {{< text bash >}}
+    $ cat <<EOF | kubectl -n "${VM_NAMESPACE}" apply -f -
+    apiVersion: v1
+    kind: Service
+    metadata:
+      name: cloud-vm
+      labels:
+        app: cloud-vm
+    spec:
+      ports:
+      - port: 7070
+        name: grpc
+      - port: 8090
+        name: http
+      selector:
+        app: cloud-vm
+    EOF
+    {{< /text >}}
+
+1. Create workload entry
 
+    {{< text bash >}}
+    $ cat <<EOF | kubectl -n "${VM_NAMESPACE}" apply -f -
+      apiVersion: networking.istio.io/v1beta1
+      kind: WorkloadEntry
+      metadata:
+        name: "${VM_NAME}"
+        namespace: "${VM_NAMESPACE}"
+      spec:
+        address: "${VM_IP}"
+        labels:
+          app: gce-vm
```

Comment:
platform-ism - please remove.. should this workloadentry not select the service name?

Reply:
Sorry on my running script I use cloud-vm. Forget to change to cloud-vm